### PR TITLE
Feature/hotel class

### PR DESCRIPTION
**Pull request by kavakai**
What (if any) features are you implementing?
Create classes and tests.
begin HTML and SCSS
What (if anything) did you refactor?
N/A
Were there any issues that arose?
Needed to utilize the Date object to use the dates properties. 
Where should the reviewer start?
classes file and corresponding tests

Refactor checklist
[] Check variable and parameter names: are they descriptive and unique?
[] Is the code DRY? SRP?
[] Is it easy for others to read your code? (explanatory comments needed?)
[] Is the syntax consistent? (e.g. use of semicolons, blank lines)
[] Are commit messages formatted correctly?
goals?
Begin event listeners in scripts.

### CODE REVIEW

File: src/Classes/User.js [hunk 1]
```diff
@@ -42,8 +42,11 @@ class User {
     sortBookings() {
         this.futureBookings = this.futureBookings.sort((bookingA, bookingB) => bookingA.date < bookingB.date);
```
Comment (Daniel-OC) [inline src/Classes/User.js]: Interesting, where and how are future and past Bookings being created?

File: test/room-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    it('should be able to keep track of the room type', function () {
+        expect(room1.type).to.equal('single room');
+        expect(room2.type).to.equal('suite');
+    });
+
+    it('should keep track if it has a bidet', function () {
+        expect(room1.hasBidet).to.equal(false);
+        expect(room2.hasBidet).to.equal(true);
+    });
+
```
Comment (Daniel-OC) [inline test/room-test.js:43]: Great, thorough testing to my eye!

File: src/Classes/Hotel.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
 class Hotel {
     constructor(rooms, bookings) {
         this.rooms = rooms;
         this.bookings = bookings;
     };
+
+    checkAvailability(dates) {
+        this.bookings.forEach(booking => {
+            let bookingDate = new Date(booking.date);
+            let requestedDate = new Date(dates);
```
Comment (Daniel-OC) [inline src/Classes/Hotel.js:12]: I'm not sure I track the logic here, would love to hear your thinking

Comment (kavakai) [reply src/Classes/Hotel.js:12]: good question Daniel. I was thinking that based on a users date selection, I would pass that date into this function... then, I would iterate through the list of all bookings to match that date with any upcoming booking dates... from there I would either confirm the booking and do my POST, or "deny" the booking with some message. I haven't really figured that all out yet, but this was my starting point. Thanks for the feedback.

**Pull request merged**